### PR TITLE
Fix tag processing

### PR DESCRIPTION
**Pull request by daenmdir**
1. changed how the new system is processed: generate tag for inactive/fresh and not active systems
2. save map prior to broadcasting it to clients
3. normalize tags before converting to int

### CODE REVIEW

File: app/Model/Pathfinder/MapModel.php [hunk 1]
```diff
@@ -508,9 +508,13 @@ public function getNewSystem(int $systemId) : SystemModel {
...
             $system->setType();
-            $system->tag = SystemTag::generateFor($system, $system, $this);
+        } elseif ($system->isActive()){
+            // system exists and is active, return as is to avoid re-tagging it
+            return $system;
         }
 
+        // generate tag for the new systems
+        $system->tag = SystemTag::generateFor($system, $system, $this);
         $system->setActive(true);
```
Comment (daenmdir) [inline app/Model/Pathfinder/MapModel.php:518]: Apparently when you edit a system (e.g. give it a tag) and remove it from PF map it goes into 'inactive' state so they will not be null at line 502, so gotta generate tag

File: app/Controller/Api/Map.php [hunk 2]
```diff
@@ -589,6 +589,11 @@ protected function updateMapsData(Pathfinder\CharacterModel $character, array $m
         }
 
         foreach($maps as $map){
+            // update Tags =================================================================================
+            $map->nextBookmarks = SystemTag::nextBookmarks($map);
+            $activeCharacter = $this->getCharacter();
+            $map->save($activeCharacter);
+
```
Comment (daenmdir) [inline app/Controller/Api/Map.php:596]: makes more sense to save it to db and only then broadcast

File: app/Lib/SystemTag.php [hunk 3]
```diff
@@ -84,10 +83,11 @@ static function intToTag(int $int): string {
      * @return int
      */
     static function tagToInt(string $tag): int {
```
Comment (Thump3d) [inline app/Lib/SystemTag.php:108]: I would change this function completely.. it still assumes 2 chars.. would also change the intToTag to match once you change it..

Comment (daenmdir) [reply app/Lib/SystemTag.php:108]: I looked into it, but not sure if it's worth the hustle. Two letters is 260 or smth variations of sigs for one class. I'd rather get punched in the dick than scan this many sigs

**Comment by MonoliYoda** [comment]
Merged onto dev branch

**Pull request closed without merging**